pkg/util/cluster.go

```diff
@@ -123,33 +121,15 @@ func (c *ClusterInfoImp) Init(ctx context.Context, cl client.Client, logger logr
 	return nil
 }
 
-func (c *ClusterInfoImp) initKubernetes(cl client.Client) error {
-	masterNodeList := &corev1.NodeList{}
-	masterReq, err := labels.NewRequirement("node-role.kubernetes.io/master", selection.Exists, nil)
-	if err != nil {
-		return err
-	}
-	masterSelector := labels.NewSelector().Add(*masterReq)
-	masterLabelSelector := client.MatchingLabelsSelector{Selector: masterSelector}
-	err = cl.List(context.TODO(), masterNodeList, masterLabelSelector)
+func (c *ClusterInfoImp) initKubernetes(ctx context.Context, cl client.Client) error {
+	var err error
+	masterNodeCount, workerNodeCount, err := getNodesCount(ctx, cl)
 	if err != nil {
 		return err
 	}
 
-	workerNodeList := &corev1.NodeList{}
-	workerReq, err := labels.NewRequirement("node-role.kubernetes.io/worker", selection.Exists, nil)
-	if err != nil {
-		return err
-	}
-	workerSelector := labels.NewSelector().Add(*workerReq)
-	workerLabelSelector := client.MatchingLabelsSelector{Selector: workerSelector}
-	err = cl.List(context.TODO(), workerNodeList, workerLabelSelector)
-	if err != nil {
-		return err
-	}
-
-	c.controlPlaneHighlyAvailable = len(masterNodeList.Items) >= 3
-	c.infrastructureHighlyAvailable = len(workerNodeList.Items) >= 2
+	c.controlPlaneHighlyAvailable = masterNodeCount >= 3
+	c.infrastructureHighlyAvailable = workerNodeCount >= 2
```

Comment:
If we allow to get async access to this fields, we MUST protect them.

Please add a RWMutext, and lock each time we read or write these fields.

alternatively, use `atomic.Bool` from the standard library, instead of bool for the `controlPlaneHighlyAvailable` and `infrastructureHighlyAvailable` fields.